Source/Blazorise/Components/Droppable/DropZone.razor.cs

```diff
@@ -107,53 +140,86 @@ private void OnContainerTransactionStarted( object sender, DraggableTransaction<
             InvokeAsync( StateHasChanged );
         }
 
-        private void OnContainerTransactionEnded( object sender, EventArgs e )
+        private void OnContainerTransactionEnded( object sender, DraggableTransactionEnded<TItem> e )
         {
-            itemOnDropZone = false;
+            dragCounter = 0;
 
             if ( GetApplyDropClassesOnDragStarted() )
             {
                 dropAllowed = false;
             }
 
+            if ( e.Success )
+            {
+                if ( e.OriginDropZoneName == Name && e.DestinationDropZoneName != e.OriginDropZoneName )
+                {
+                    indices.Remove( e.Item );
+                }
+
+                if ( e.OriginDropZoneName == Name || e.DestinationDropZoneName == Name )
+                {
+                    int index = 0;
+
+                    foreach ( var item in indices.OrderBy( x => x.Value ).ToArray() )
```

Comment:
Decison about Dictionary<TItem, int> instead of Dictionary<int, TItem> should the int be the actual key and the value, the TItem?

Wouldn't the index always be unique?
Would we ever have two 'equal' TItem? Or is that a dumb assumption?

Would that make the implementation harder? To reorder the items?

Reply:
TItem is the key because the TItem is the one that can have a different index when reordered. Kinda make more sense to me.